app/services/rag_wise/tools/course_forum_discussions_tool.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+class RagWise::Tools::CourseForumDiscussionsTool
+  extend Langchain::ToolDefinition
+
+  define_function :get_discussions,
+                  description: 'Retrieve past course forum discussions that are semantically closest\
+                  to the user query.' do
+    property :user_query, type: 'string', description: 'Exact user query', required: true
+  end
+
+  def initialize(course, evaluation)
+    @client = LANGCHAIN_OPENAI
+    @course = course
+    @evaluation = evaluation
+  end
+
+  def get_discussions(user_query:)
+    query_embedding = @client.embed(text: user_query, model: 'text-embedding-ada-002').embedding
+    data = @course.nearest_forum_discussions(query_embedding)
```

Comment:
consider allowing `nearest_forum_discussion` to support a parameter `k` to control the number of matching neighbours

Reply:
Introduced limit parameter